db/tests/types/test_alteration.py

```diff
@@ -327,6 +336,19 @@
                 VALID: [("432", 432)],
                 INVALID: ["1.2234"]
             },
+            DATE: {
+                VALID: [
+                    ("1999-01-18", date(1999, 1, 18)),
+                    ("1/18/1999", date(1999, 1, 18)),
+                    ("jan-1999-18", date(1999, 1, 18)),
+                    ("19990118", date(1999, 1, 18)),
+                ],
+                INVALID: [
+                    "18/1/1999",
```

Comment:
This is a valid date in a non-US format (this is how I grew up writing dates). Does Postgres not parse this?

Reply:
This is dependent on the server [`DateStyle`](https://www.postgresql.org/docs/13/runtime-config-client.html#GUC-DATESTYLE) option, which defaults to MDY. As its a server wide setting, as of now I don't think we can support parsing multiple different formats. If it is something we want, we might be able to do it with a custom type that has a `format` type option? I'm dealing with a similar problem for the `TIME` type right now, so I'll know how hard it'll be once that's done.

Reply:
Okay, let's not worry about it now, we can open a new issue for improving parsing later if needed.

I'll review again once there's a test for ensuring that dates are inferred correctly on import. Thanks!

Reply:
Also please add a comment in the code about the `DateStyle` option.